gc/base/Heap.cpp

```diff
@@ -544,8 +544,10 @@ MM_Heap::getActualSoftMxSize(MM_EnvironmentBase* env)
 
 		uintptr_t nurserySize = totalHeapSize - tenureSize;
 
-		if (nurserySize <= extensions->softMx) {
+		if (MEMORY_TYPE_OLD == memoryType && nurserySize <= extensions->softMx) {
 			actualSoftMX = extensions->softMx - nurserySize;
+		} else if (MEMORY_TYPE_NEW == memoryType) {
+			actualSoftMX = extensions->softMx / 4;
```

Comment:
We cannot hard code / 4. It should be Xmnx/Xmx (what does default to 1/4)

Reply:
i'd reorg this code:

```
if new type 
     softMx * Xmnx/Xmx
else if old type
     if nursery size <= soft mx
         softMx - nursery size
     else
         0
else
     unreachable if any other type passed
```

Reply:
should the line 
`actualSoftMX  = softMx * Xmnx/Xmx`

be 

`actualSoftMX = (softMx * Xmnx/Xmx) - nurserySize` ?

Otherwise, `actualSoftMx` will always be equal to the same thing for the entire run for `MEMORY_TYPE_NEW`, and could lead to a large imbalance between nursery and tenure

Reply:
Hm, despite you explanation, I don't understand a potential danger and motivation for `...- nurserySize`

I'll take an example, let's say we have specified on command line -Xsoftmx2G -Xmnx4G -Xmx4G. And let's say softmx is still 2G, and current heap size is 1GB, split equally at 500M between tenure and nursery.

If nursery is under pressure and wants to grow (but still maintain softmx 2G), we will continue growing it till Xmnx/Xmx*softmx = 4G/8Gx2G = 1G. So, it will leave guarantied 1G for Tenure  (extra 500M on top of current 500M) to grow if ever needed in future.

So, Nursery can be larger than tenure, if Xmnx/Xmx ratio allows it (in this example it, allowed to be as much as equal, but it can be even larger, for large enough  Xmnx/Xmx ratio).

Just to complete the scenario (and understanding how tenure and nursery can grow), for the same example, if tenure is under pressure, it will be able to grow to total of softmx - 'current nursery' =  2G - 500M = 1.5G. If it indeed does so, unlike the previous example, the nursery will be 'stuck' and won't be able grow further than those current 500M. It's still all ok.